lib/vehicle.rb

```diff
@@ -5,21 +5,40 @@ class Vehicle
               :year,
               :make,
               :model,
-              :engine
+              :engine,
+              :registered,
+              :plate_type,
+              :registration_date
 
   def initialize(vehicle_details)
     @vin = vehicle_details[:vin]
     @year = vehicle_details[:year]
     @make = vehicle_details[:make]
     @model = vehicle_details[:model]
     @engine = vehicle_details[:engine]
+    @registered = false
+    @registration_date = nil
   end
 
   def antique?
-    Date.today.year - @year > 25
+    antique = Date.today.year - @year > 25
   end
 
   def electric_vehicle?
     @engine == :ev
   end
+
+  def plate_type
+    if @engine == :ev
+      @plate_type = :ev
+    elsif antique?
+      @plate_type = :antique
+    else 
+      @plate_type = :regular
```

Comment:
Yes, see, you have the @plate_type attribute here, but it's missing from the initialize so it won't ever get set.